### PR TITLE
Allow Fleet provision to run together with iotcored

### PR DESCRIPTION
**Pull request by rawalexe**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: docs/examples/supported_lifecyle_types/1.json [hunk 1]
```diff
@@ -0,0 +1,53 @@
+{
+  // Notice: All Key's are case sensitive
```
Comment (patrzhan) [inline docs/examples/supported_lifecyle_types/1.json:2]: I don't think json supports comments like this (github also highlights it in red)

File: fleet-provisioning/src/entry.c [hunk 2]
```diff
@@ -6,23 +6,27 @@
...
 #include <openssl/types.h>
 #include <openssl/x509.h>
-#include <string.h>
 #include <stdint.h>
 #include <stdio.h>
 
 #define MAX_TEMPLATE_LEN 129
 #define MAX_ENDPOINT_LENGTH 129
 #define MAX_TEMPLATE_PARAM_LEN 4096
+#define MAX_PATH_LEN 4096
```
Comment (patrzhan) [inline fleet-provisioning/src/entry.c:29]: Use PATH_MAX?

Comment (rawalexe) [reply fleet-provisioning/src/entry.c:29]: wanted to use custom here on purpose

File: fleet-provisioning/src/entry.c [hunk 3]
```diff
@@ -187,8 +191,98 @@ static GglError fetch_from_db(FleetProvArgs *args) {
     return GGL_ERR_OK;
 }
 
-GglError run_fleet_prov(FleetProvArgs *args) {
-    GglError ret = fetch_from_db(args);
+static GglError update_cred_access(void) {
+    char *args[] = { "chown", "-R", "ggcore:ggcore", "/ggcredentials/", NULL };
```
Comment (patrzhan) [inline fleet-provisioning/src/entry.c:195]: While ggcore is our default, it is configurable by customer with GGL_SYSTEMD_SYSTEM_USER. So this doesn't work on any install that has configured the user to be something different.

Comment (rawalexe) [reply fleet-provisioning/src/entry.c:195]: if they are building it from the scratch then they will need to update this section also as there is no way for me to pull in cmake  configurations

File: fleet-provisioning/src/entry.c (hunk 2 above)
Comment (patrzhan) [reply fleet-provisioning/src/entry.c:29]: Why?

File: fleet-provisioning/src/entry.c (hunk 3 above)
Comment (patrzhan) [reply fleet-provisioning/src/entry.c:195]: Sounds good, maybe we can bring this out to be a macro so if they do want to configure this it's not buried too deep in the code

Comment (rawalexe) [reply fleet-provisioning/src/entry.c:195]: Added a warning in the docs

**Review by patrzhan — APPROVED** [review]

**Pull request merged**